src/dinit.cc

```diff
@@ -453,6 +471,68 @@ static int process_commandline_arg(char **argv, int argc, int &i, options &opts)
     return 0;
 }
 
+#if SUPPORT_SELINUX
+static bool selinux_transition(const char *exe) {
```

Comment:
I would like a reasonably extensive comment on this explaining what it does (maybe include any relevant links to SElinux docs), and why it needs to be done early. The purpose of the `exe` argument should be explained. The fact that if this function returns an error (false) then dinit should terminate should be mentioned.